lib/req/steps.ex

```diff
@@ -285,6 +285,7 @@ defmodule Req.Steps do
       data = request.options[:json] ->
         %{request | body: Jason.encode_to_iodata!(data)}
         |> Req.Request.put_new_header("content-type", "application/json")
+        |> Req.Request.put_new_header("accept", "application/json, */*;q=0.5")
```

Comment:
does curl also set `*/*` in these circumstances? If not let's not do that yet.

Reply:
`curl` actually only sets `*/*`. I took the `application/json, */*;q=0.5` example from [httpie](https://github.com/httpie/httpie).

## curl
```
curl -v -X POST https://reqbin.com/echo/post/json -H 'Content-Type: application/json' -d '{"hello":"world"}'

> POST /echo/post/json HTTP/2
> Host: reqbin.com
> user-agent: curl/7.79.1
> accept: */*
> content-type: application/json
> content-length: 17
```

## httpie
```
http -v PUT pie.dev/put name=John email=john@example.org

Accept: application/json, */*;q=0.5
Accept-Encoding: gzip, deflate
Connection: keep-alive
Content-Length: 45
Content-Type: application/json
Host: pie.dev
User-Agent: HTTPie/3.1.0
```

Reply:
sorry, I specifically meant the `curl --json` option. It sets application/json:

```
curl --json '{ "drink": "coffe" }' https://httpbin.org/anything
{
  "args": {},
  "data": "{ \"drink\": \"coffe\" }",
  "files": {},
  "form": {},
  "headers": {
    "Accept": "application/json",
    "Content-Length": "20",
    "Content-Type": "application/json",
    "Host": "httpbin.org",
    "User-Agent": "curl/7.83.0",
    "X-Amzn-Trace-Id": "Root=1-62b32ce1-34c3c2874e099e1c361c5b28"
  },
  "json": {
    "drink": "coffe"
  },
  "method": "POST",
  "origin": "91.150.170.123",
  "url": "https://httpbin.org/anything"
}
```

let's go with that but I'm happy to adjust down the road.